src/test/java/com/newrelic/agent/instrumentation/verify/AfterEvaluationActionTest.java

```diff
@@ -0,0 +1,138 @@
+
+    @BeforeEach
+    void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+        this.getRepositoryFunction = project -> Collections.emptyList();
+        this.testClass = new AfterEvaluationAction(mockOptions, mockTask, mockLogger, mockDestinationDir, getRepositoryFunction);
+    }
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    public Project mockProject;
+
+    @Mock
+    public Task mockTask;
+
+    @Mock
+    public File mockDestinationDir;
+
+    @Mock
+    public Logger mockLogger;
+
+    @Mock
+    public VerifyInstrumentationOptions mockOptions;
+
+
+    @Test
+    void shouldVerifyProjectWithTaskNameVerifyInstrumentation() {
+        String projectPath = "/pathToProjectBuildFile";
+        List<String> taskNames = new ArrayList<>();
+        taskNames.add(VERIFIER_TASK_NAME);
+        when(mockProject.getGradle().getStartParameter().getTaskNames()).thenReturn(taskNames);
+        when(mockProject.getGradle().getStartParameter().getCurrentDir().getPath()).thenReturn(projectPath);
+        when(mockProject.getProjectDir().getPath()).thenReturn(projectPath);
+
+        assertTrue(AfterEvaluationAction.projectRequiresVerification(mockProject));
+    }
+
+    @Test
+    void shouldVerifyProjectWithProjectPathBeforeTaskName() {
+        String projectPath = "/pathToProject/";
+        List<String> taskNames = new ArrayList<>();
+        taskNames.add(projectPath+VERIFIER_TASK_NAME);
+        when(mockProject.getGradle().getStartParameter().getTaskNames()).thenReturn(taskNames);
+        when(mockProject.project(":"+projectPath).getProjectDir().getPath()).thenReturn(projectPath);
+        when(mockProject.getProjectDir().getPath()).thenReturn(projectPath);
+
+        assertTrue(AfterEvaluationAction.projectRequiresVerification(mockProject));
+    }
+
+    @Test
+    void shouldNotVerifyProjectDueToTaskName() {
+        List<String> taskNames = new ArrayList<>();
+        taskNames.add(VERIFIER_TASK_NAME + "needs to be at end. This won't work.");
+        when(mockProject.getGradle().getStartParameter().getTaskNames()).thenReturn(taskNames);
+
+        assertFalse(AfterEvaluationAction.projectRequiresVerification(mockProject));
+    }
+
+    @Test
+    void shouldNotVerifyProjectWhenNoProjectWithPath() {
+        String projectPath = "/noProjectAtPath";
+        List<String> taskNames = Arrays.asList(projectPath + VERIFIER_TASK_NAME);
+
+        when(mockProject.getGradle().getStartParameter().getTaskNames()).thenReturn(taskNames);
+        when(mockProject.project(":"+projectPath)).thenThrow(new UnknownProjectException("nothing here"));
+
+        assertFalse(AfterEvaluationAction.projectRequiresVerification(mockProject));
+    }
+
+    @Test
+    void shouldCreateDependencyOnAgentOfFileType() {
+        File mockAgent = new File("/agentForTest");
+        Project testProject = ProjectBuilder.builder().build();
+
+        testClass.createProjectDependencyOnAgent(testProject, mockAgent);
+
+        assertEquals(1,
```

Comment:
I couldn't find a more precise way to verify the dependency is the mockAgent. That is why I'm verifying by count.